src/app/core/device.ts

```diff
@@ -820,47 +710,45 @@ export default class Device {
      * @memberof Device
      * @returns {void}
      */
-    setYSpan(value) {
-        this.params.updateParameter("length", value);
+    setYSpan(value: number): void {
+        this.__params.updateParameter("length", value);
     }
 
     /**
      * Returns the X-Span Value
      * @return {number}
      * @memberof Device
      */
-    getXSpan() {
-        return this.params.getValue("width");
+    getXSpan(): number {
+        return this.__params.getValue("width");
     }
 
     /**
      * Returns the Y-Span Value
      * @return {number}
      * @memberof Device
      */
-    getYSpan() {
-        return this.params.getValue("length");
+    getYSpan(): number {
+        return this.__params.getValue("length");
     }
 
     /**
      * Create the layers necessary for creating a new level
      * @return {Array<Layer>} Returns a the layer objects created
      * @memberof Device
      */
-    createNewLayerBlock() {
-        const flowlayer = new Layer({ z_offset: 0, flip: false }, "flow" + this.__layerBlockIndex.toString(), "FLOW", this.__layerBlockIndex.toString());
-        const controllayer = new Layer({ z_offset: 0, flip: false }, "control" + this.__layerBlockIndex.toString(), "CONTROL", this.__layerBlockIndex.toString());
-        // TODO: remove cell layer from the whole system
-        const cell = new Layer({ z_offset: 0, flip: false }, "cell" + this.__layerBlockIndex.toString(), "FLOW", this.__layerBlockIndex.toString());
+    createNewLayerBlock(): Array<Layer> {
+        let flowlayer = new Layer({ z_offset: 0, flip: false }, "flow");
+        let controllayer = new Layer({ z_offset: 0, flip: false }, "control");
+        //TODO: remove cell layer from the whole system
+        let cell = new Layer({ z_offset: 0, flip: false }, "cell");
```

Comment:
Yeah the missing arguments here will need to be enabled

Reply:
Do you mean that z_offset and flip should be arguments rather than set within the function?